components/ledger/pkg/ledger/resolver.go

```diff
@@ -104,10 +107,13 @@ func ProvideResolverOption(provider interface{}) fx.Option {
 func ResolveModule(cacheBytesCapacity, cacheMaxNumKeys int64) fx.Option {
 	return fx.Options(
 		fx.Provide(
-			fx.Annotate(func(storageFactory storage.Driver[Store], ledgerOptions []LedgerOption, options ...ResolverOption) *Resolver {
-				return NewResolver(storageFactory, ledgerOptions, cacheBytesCapacity, cacheMaxNumKeys, options...)
-			}, fx.ParamTags("", ResolverLedgerOptionsKey, ResolverOptionsKey)),
+			fx.Annotate(func(storageFactory storage.Driver[Store], ledgerOptions []LedgerOption, locker Locker, options ...ResolverOption) *Resolver {
+				return NewResolver(storageFactory, ledgerOptions, cacheBytesCapacity, cacheMaxNumKeys, locker, options...)
+			}, fx.ParamTags("", ResolverLedgerOptionsKey, "", ResolverOptionsKey)),
 		),
+		fx.Provide(func() Locker {
+			return NewInMemoryLocker()
```

Comment:
Ok, so no redis anymore since we are in monowriter ? Should we keep it in case we need it one day ?

Reply:
I don't think is is necessary.
If we use Redis later, the implementation for account based lock will be more complicated.